contracts/arbitrage/BancorArbitrage.sol

```diff
@@ -0,0 +1,479 @@
+        ISwapRouter initUniswapV3Router,
+        IUniswapV2Router02 initSushiSwapRouter
+    )
+        validAddress(address(initBnt))
+        validAddress(address(initBancorNetworkV2))
+        validAddress(address(initBancorNetworkV3))
+        validAddress(address(initUniswapV2Router))
+        validAddress(address(initUniswapV3Router))
+        validAddress(address(initSushiSwapRouter))
+    {
+        _bnt = initBnt;
+        _weth = IERC20(initUniswapV2Router.WETH());
+        _bancorNetworkV2 = initBancorNetworkV2;
+        _bancorNetworkV3 = initBancorNetworkV3;
+        _uniswapV2Router = initUniswapV2Router;
+        _uniswapV3Router = initUniswapV3Router;
+        _sushiSwapRouter = initSushiSwapRouter;
+    }
+
+    /**
+     * @dev fully initializes the contract and its parents
+     */
+    function initialize() external initializer {
+        __BancorArbitrage_init();
+    }
+
+    // solhint-disable func-name-mixedcase
+
+    /**
+     * @dev initializes the contract and its parents
+     */
+    function __BancorArbitrage_init() internal onlyInitializing {
+        __ReentrancyGuard_init();
+        __Upgradeable_init();
+
+        __BancorArbitrage_init_unchained();
+    }
+
+    /**
+     * @dev performs contract-specific initialization
+     */
+    function __BancorArbitrage_init_unchained() internal onlyInitializing {
+        _rewards = Rewards({ percentagePPM: 100000, maxAmount: 100 * 1e18 });
+    }
+
+    /**
+     * @dev authorize the contract to receive the native token
+     */
+    receive() external payable {}
+
+    /**
+     * @inheritdoc Upgradeable
+     */
+    function version() public pure override(Upgradeable) returns (uint16) {
+        return 1;
+    }
+
+    /**
+     * @dev checks whether the specified number of routes is supported
+     */
+    modifier validRouteLength(Route[] calldata routes) {
+        // validate inputs
+        _validRouteLength(routes);
+
+        _;
+    }
+
+    function _validRouteLength(Route[] calldata routes) internal pure {
+        if (routes.length == 0 || routes.length > MAX_ROUTE_LENGTH) {
+            revert InvalidRouteLength();
+        }
+    }
+
+    /**
+     * @dev returns true if given token is WETH
+     */
+    function _isWETH(Token token) internal view returns (bool) {
+        return address(token) == address(_weth);
+    }
+
+    /**
+     * @dev sets the rewards parameters
+     *
+     * requirements:
+     *
+     * - the caller must be the admin of the contract
+     */
+    function setRewards(
+        Rewards calldata settings
+    ) external onlyAdmin validFee(settings.percentagePPM) greaterThanZero(settings.maxAmount) {
+        uint32 prevPercentagePPM = _rewards.percentagePPM;
+        uint256 prevMaxAmount = _rewards.maxAmount;
+
+        // return if the settings are the same
+        if (prevPercentagePPM == settings.percentagePPM && prevMaxAmount == settings.maxAmount) {
+            return;
+        }
+
+        _rewards = settings;
```

Comment:
setRewards -> updateSettings
 _rewards = settings > _settings = newSettings
 
 and so on everywhere else

Reply:
see comment above re. explicitness